contribute/localisation.md

```diff
@@ -0,0 +1,175 @@
+# Localisation
+
+Grafana uses the [LinguiJS](https://github.com/lingui/js-lingui) framework for managing translating phrases in the Grafana frontend.
+
+## tl;dr
+
+- Use `<Trans id="search-results.panel-link">Go to {panel.title}</Trans>` in code to add a translatable phrase
+- Translations are stored in .po files in `public/locales/{locale}/messages.po`
+- If a particular phrase is not available in the a language then it will fall back to English
+
+## How to add a new translation phrase
+
+1. Use one of `@lingui/macro`'s React components with the `id`, ensuring it conforms to the guidelines below, with the default english translation. e.g.
+
+```jsx
+import { Trans } from @lingui/macro
+
+const SearchTitle = ({term}) => (
+  <Trans id="search-page.results-title">
+    Results for {term}
+  </Trans>
+);
+```
+
+Prefer using the JSX components (compared to the plain javascript functions, see below) where possible for phrases. Many props can (and probably should) be changed to accept the `React.ReactNode` instead of `string` for phrases put into the DOM.
+
+Note that Lingui must be able to statically analyse the code to extract the phrase, so the `id` can not be dynamic. e.g. the following will not work:
+
+```jsx
+const ErrorMessage = ({ id, message }) => <Trans id={`errors.${id}`}>There was an error: {message}</Trans>;
+```
+
+2. Upon reload, the default English phrase will appear on the page.
+
+3. Before submitting your PR, run the `yarn i18n:extract` command to extract the messages you added into the `messages.po` file and make them available for translation.
+
+## How translations work in Grafana
+
+Grafana uses the [LinguiJS](https://github.com/lingui/js-lingui) framework for managing translating phrases in the Grafana frontend. We use Lingui to mark up phrases for extraction, to automate extracting phrases into message catalogs for use in translation management systems, and then to load the translated phrases into the frontend for the user's selected locale.
```

Comment:
🖌️ I think in this section would be awesome if we get a diagram that describes this flow  :)

Reply:
I used this in the past :) https://mermaid-js.github.io/mermaid/#/